### DISCUSSION TITLE
Alternate or backup storage location

### DISCUSSION THREAD

**Discussion by bubba925** [blakeblackshear/frigate#8997] [Ideas]
I was thinking about this today and it’s probably a pretty niche scenario but figured I’d throw it out there. I run Frigate on Proxmox and use Unraid as my storage. They are on separate UPSs with the Unraid server having a shorter runtime. I was curious if it is possible to somehow, in the event my primary storage is inaccessible it would fail to another internal location? Probably not worth the effort to make it fully functional and reliable. I currently have a script set to reboot the Frigate VM on array start to prevent stale files and network drive issues.

**Comment by PyrokineticDarkElf** [comment]
I've got similar requirements. I can see a lot of possibilities here too. For example, failover to internal storage or a smaller NAS or even switch over to cloud storage. Perhaps when Frigate notices, it could prioritise some cameras over others.

Presumably, when the main storage location comes back online, any data on the backup location would also need to be moved.

I think this could be pretty useful!

**Comment by hagak** [comment]
I would think this would be better solved external to frigate.  For one Frigate can only know about the storage AFTER the fact, ie it gets an error when trying to write something.

The problem you are trying to solve is a pretty complex one, off the top of my head the way I would first look is using iSCSI multipathing however this is more for making the network path redundant NOT the backend storage.   Another solution could be something like CEPHFS or Longhorn.  These are basically distributed file systems that can stay online in the case of full outage of a node, however this is not a lightweight solution.
